mailbox/maildir/src/main/java/org/apache/james/mailbox/maildir/MaildirFolder.java

```diff
@@ -765,7 +777,7 @@ public Long execute() throws MailboxException {
                         while ((line = reader.readLine()) != null)
                             lines.add(line);
                         uid = getNextUid();
-                        lines.add(String.valueOf(uid) + " " + name);
+                        lines.add(String.valueOf(uid.asLong()) + " " + name);
```

Comment:
I think it's not a good idea because uid is never defined as a string and you have no way to build on from a string.